kyaml/fn/framework/validation.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package framework
+
+import (
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+// SchemaFromFunctionDefinition extracts the schema for a particular GVK from the provided KRMFunctionDefinition
+// Since the relevant fields of KRMFunctionDefinition exactly match the ones in CustomResourceDefinition,
+// this helper can also load CRDs (e.g. produced by KubeBuilder) transparently.
+func SchemaFromFunctionDefinition(gvk resid.Gvk, data string) (*spec.Schema, error) {
+	var def KRMFunctionDefinition
+	// need to use sigs yaml because spec.Schema type only has json tags
```

Comment:
FYI I'm expecting this to be fixed soon https://github.com/kubernetes/kube-openapi/pull/279

So we can go back and clean up the places we're using sigs yaml after that PR is merged.

Reply:
That's great! There will still be a few places we need to do it, to support decoding into k/k types and types using metav1 metadata. But the fewer, the better!